test/runtests.jl

```diff
@@ -82,6 +82,10 @@ end
   a = S(1; exponent=(2 * i * j) * 1//R((q - 1)))
   @test isone(a^(q-1))
   @test a == a^q
+
+  @test E(1)^(1//2) == E(2)
+  @test E(2)^(1//3) == E(6)
+  @test E(3)^(1//2) == E(6)
```

Comment:
I am conflicted about this. Last time we talked about it, didn't we agree that `E(n)^(p//q)` should be allowed (it throws a `MethodError` right now), but that the values returned by `E(n)^R(p//q)` are otherwise fine. 

In particular I dislike these values. I'd really expect this:
```suggestion
  @test E(1)^(1//2) == E(1) ==  1
  @test E(2)^(1//3) == E(2) == -1
  @test E(3)^(1//2) == E(6)
```

However, I guess a strong argument for your choices is consistency with `QQAbField`:
```
julia> root(E(1), 2)
-1

julia> root(E(2), 3)
ζ(6)

julia> root(E(3), 2)
ζ(6)

julia> E(6)
-ζ(3)        # huh???
```

Then again, `E(3)^(1//2)` does not work there. And `root(x,n::Int)` is documented as 
> Return the principal `n`-th root of `x`. We require `x \geq 0`.


Aha but `QQBar` closes the loop:
```
julia> L = algebraic_closure(QQ)
Algebraic closure of rational field

julia> L(E(3))
Root -0.500000 + 0.866025*im of x^2 + x + 1

julia> L(E(3))^(1//2)
Root 0.500000 + 0.866025*im of x^2 - x + 1

julia> L(root(E(3),2))
Root 0.500000 + 0.866025*im of x^2 - x + 1
```


So I guess we need this patch for consistency.